editor/gui/editor_file_dialog.cpp

```diff
@@ -1826,11 +1839,15 @@ EditorFileDialog::EditorFileDialog() {
 	makedir->connect("pressed", callable_mp(this, &EditorFileDialog::_make_dir));
 	pathhb->add_child(makedir);
 
-	list_hb = memnew(HSplitContainer);
+	body_hbox = memnew(HBoxContainer);
```

Comment:
I think this should be a split container as well, and the ratio argument should be removed. You can already control the minimum size from the control itself. If you really need to adjust the default split so it can be, say, bigger than the min size, then the stretch ratio can be replaced for the split ratio/offset.

Reply:
That's a good point, but it does prevent adding multiple side menus. I only need one personally, but I wanted to mention it.

Reply:
I think that we don't need to worry about such case, besides you can add any control you want with the API you expose. So if ever the need arises, we can create the necessary layout outside of the EditorFileDialog.

But you also point out an important detail. We should probably track the added control and err if you try to add more than one. And this can be mentioned in the docs.